src/rust/src/contract.rs

```diff
@@ -286,6 +302,20 @@ pub(crate) fn preflight_host_function(
     hf_buf: &CxxVector<u8>,
     args_buf: &CxxVector<u8>,
     cb: UniquePtr<PreflightCallbacks>,
+) -> Result<(), Box<dyn Error>> {
+    let res = panic::catch_unwind(panic::AssertUnwindSafe(|| {
+        preflight_host_function_or_maybe_panic(hf_buf, args_buf, cb)
+    }));
+    match res {
+        Err(_) => Err(ContractError::General("contract host panicked").into()),
```

Comment:
It is somewhat confusing that we have `ContractError` in the host which is host errors, as well as `Status::ContractError` which is only contract generated errors. This is my fault, I didn't see that we had `ContractError` here when I added the new variant to `Status`.

If we are to change either, it might be preferable if we can change the `ContractError` though, but maybe we can live with name clash.

Reply:
Oh, I can change this, not a big deal.